jira-slack-server-integration/jira-slack-server-integration-plugin/src/main/java/com/atlassian/jira/plugins/slack/service/listener/SlackEventHandlerService.java

```diff
@@ -300,11 +299,7 @@ private void sendMessageToDedicatedChannel(final DedicatedChannel dedicatedChann
                 null, Type.DEDICATED));
 
         final JiraCommandEvent event = new IssueMentionedEvent(message, dedicatedChannel.getIssueId());
-        final SendNotificationTask task = taskBuilder.newSendNotificationTask(
-                event,
-                notificationInfo,
-                taskExecutorService);
-        taskExecutorService.submitTask(task);
+        asyncExecutor.run(taskBuilder.newSendNotificationTask(event, notificationInfo, asyncExecutor));
```

Comment:
NIT: Creating a separate variable for `taskBuilder.newSendNotificationTask(event, notificationInfo, asyncExecutor)` and passing it to the parameter argument would have been more precise

Reply:
You are right. Having a task in a separate variable may be helpful during the debug process.